src/nrnoc/eion.cpp

```diff
@@ -75,6 +75,9 @@ void ion_register(void) {
     if (s && s->type == MECHANISM && memb_func[s->subtype].alloc == ion_alloc) {
         hoc_symlist = sav;
         free(buf);
+        if (*getarg(2) != global_charge(s->subtype)) {
+            hoc_execerr_ext("%s already defined with charge %g, cannot redefine with charge %g", s->name, global_charge(s->subtype), *getarg(2));
```

Comment:
This is a good fix, I don't understand why there are two error messages, but I do prefer this one (as it gives a python error and doesn't exit). Is it worth copying it over to replace the calcium one (line 203)?

Reply:
I agree the situation is not completely satisfying.  If there is an nmodl mechanism with a ```USEION ca ...``` statement, then ```ca_ion``` will already be defined and a bad charge value with ```h.ion_register("ca", charge)``` will give the nice error message implemented by this pull request. The case of the ugly original message followed by an exit, can only happen if there are no loaded mod files that ```USEION ca ...``` and the first definition of ca is with a ```h.ion_register("ca", charge_value_that_is_not_2)```.  (The only reason that na and k ions don't have this issue is because hh is a built-in mechanism which is loaded before the interpreter starts and so na and k ions are always defined.)  It would certainly be possible to always check early in ```h.ion_register``` whether ca is being declared, and, if so, then if the charge is not 2,
emit the error. I will go ahead and do that.